roles/ceph-facts/tasks/facts.yml

```diff
@@ -308,30 +308,36 @@
     path: '/etc/ceph/{{ cluster }}.conf'
   register: ceph_conf
 
+- name: set default osd_pool_default_crush_rule fact
+  set_fact:
+    osd_pool_default_crush_rule: "{{ ceph_osd_pool_default_crush_rule }}"
+
 - name: get default crush rule value from ceph configuration
-  command: grep 'osd pool default crush rule' /etc/ceph/{{ cluster }}.conf
-  register: crush_rule_variable
-  changed_when: false
-  check_mode: no
-  failed_when: false
+  block:
+    - &read-osd-pool-default-crush-rule
+      name: read osd pool default crush rule
+      command: grep 'osd pool default crush rule' /etc/ceph/{{ cluster }}.conf
+      register: crush_rule_variable
+      changed_when: false
+      check_mode: no
+      failed_when: crush_rule_variable.rc not in (0, 1)
+    - &set-osd-pool-default-crush-rule-fact
+      name: set osd_pool_default_crush_rule fact
+      set_fact:
+        osd_pool_default_crush_rule: "{{ crush_rule_variable.stdout.split(' = ')[1] }}"
+      when: crush_rule_variable.rc == 0
   when: ceph_conf.stat.exists | bool
 
 - name: get default crush rule value from running monitor ceph configuration
-  command: grep 'osd pool default crush rule' /etc/ceph/{{ cluster }}.conf
-  register: crush_rule_variable
-  changed_when: false
-  check_mode: no
-  failed_when: false
-  run_once: true
-  delegate_to: "{{ running_mon }}"
+  block:
+    - <<: *read-osd-pool-default-crush-rule
```

Comment:
@BenoitKnecht I'm a bit confused, I think I'm missing something.
Why would we run the task `read osd pool default crush rule` if we have a condition `- not ceph_conf.stat.exists | bool` in that block?

Reply:
That's the way it was implemented before, I didn't change the logic there.

It makes sense because the task is delegated to `running_mon` in this case (line 335 below).

Maybe it's the YAML override syntax that isn't clear here. The

```yaml
<<: *anchor
option: value
```

syntax simply takes all the keys referred to by `*anchor`, and adds (or overrides if it was already present in `*anchor`) the `option` key.

So here, we run the same command as in the `read osd pool default crush rule` task from the block above (line 317), but with

```yaml
run_once: true
delegate_to: "{{ running_mon }}"
```

added.

Reply:
@guits Have you had a chance to take another look at this? Is it clearer now, or did I misunderstand your question?